src/features/identityCheck/utils/useEduConnectLogin.ts

```diff
@@ -1,51 +1,44 @@
 import { useCallback, useEffect, useState } from 'react'
 
 import { EduConnectError } from 'features/identityCheck/errors/eduConnect/types'
-import { EduConnectErrorMessageEnum } from 'features/identityCheck/errors/hooks/useNotEligibleEduConnectErrorData'
 import { eduConnectClient } from 'libs/eduConnectClient'
 
 export function useEduConnectLogin() {
   const [loginUrl, setLoginUrl] = useState<string>()
-  const [error, setError] = useState<EduConnectError | null>(null)
+  const [error, setError] = useState<EduConnectError | Error | null>(null)
```

Comment:
la définition actuelle d'`EduConnectError`

```ts
export class EduConnectError extends Error {
```

donc on peut simplifier en
```suggestion
  const [error, setError] = useState<Error | null>(null)
```

---

je recommande d'éviter les `null` et préféré les `undefined` mais c'est peut etre un autre refactoring